src/+replab/+fp/CosetTable.m

```diff
@@ -92,15 +92,81 @@ function cosetEnumerationR(self)
                 invind = ind - nG;
             end
         end
+        
+        function compress(self)
+        % Removes repeated cosets from table
+        %
+        % COMPRESS, Holt p. 167
+            g = 0;
+            for alpha = 1:self.n
+                if self.p(alpha) == alpha
+                    g = g + 1;
+                    if g ~= alpha
+                        for x = 1:self.nGenerators*2
+                            beta = self.C(alpha, x);
+                            if beta == alpha
+                                beta = g;
+                            end
+                            self.C(g, x) = beta; 
+                            self.C(beta, self.generatorInverse(x)) = g;
+                        end
+                    end
+                end
+            end
+            self.n = g;
+            self.p = 1:self.n;
+            self.C = self.C(1:self.n, :);
+        end
+        
+        function switchElmts(self, beta, g)
+        % Switch the elements beta and g
+        %
+        % SWITCH, Holt p. 167
+            for x = 1:self.nGenerators*2
+                z = self.C(g, x);
+                self.C(g, x) = self.C(beta, x);
+                self.C(beta, x) = z;
+                for alpha = 1:self.n
+                    if self.C(alpha, x) == beta
+                        self.C(alpha, x) = g;
+                    elseif self.C(alpha, x) == g
+                        self.C(alpha, x) = beta;
+                    end
+                end
+            end
+        end
+        
+        function standardize(self)
+        % Standardizes coset table such that elements appear in order they
+        % would with no coincidences
+        %
+        % STANDARDIZE, Holt, p. 168
+            g = 2;
+            for alpha = 1:self.n
+                for x = 1:self.nGenerators*2
+                    beta = self.C(alpha, x);
+                    if beta >= g
+                        if beta > g
+                            self.switchElmts(g, beta)
+                        end
+                        g = g + 1;
+                        if g == self.n
+                            return
+                        end
+                    end
+                end
+            end
+        end
+                            
 
         function define(self, alpha, x)
         % Defines a new coset
         %
         % DEFINE, Holt p. 153
         %
         % Args:
-        %   alpha (integer): TODO
-        %   x (integer): TODO
+        %   alpha (integer): next unused coset number
```

Comment:
Is it? or is it the coset that x maps to, where the new coset number is derived from `n+1`?

Reply:
Corrected in develop.